### ISSUE TITLE
[Bug]: app.removeUser does not logout the user

### ISSUE THREAD

**Issue by Navil** [realm/realm-dart#880]
### What happened?

When calling
`await app.removeUser(user);`

the user is supposed to be logged out in the process. This does however not happen.

`await user.logOut();`

needs to be called afterwards to logout the user.

### Repro steps

1.) Login with any provider
2.) Call app.remoteUser(user)
3.) The user is still logged in but removed from the device 

### Version

Flutter (Channel stable, 3.3.0, on macOS 12.5.1 21G83 darwin-x64, locale en-GB)

### What Realm SDK flavor are you using?

MongoDB Atlas (i.e. Sync, auth, functions)

### What type of application is this?

Flutter Application

### Client OS and version

0.4.0+beta

### Code snippets

`logout(bool? deleteUser) async {
    if (user == null) {
      return;
    }
    if (deleteUser == true) {
      await app.removeUser(user!);
    }
    /*
    This line belongs in the else-block, is however needed as removeUser does
    not properly sign out.
    */
    await user?.logOut();

    //Logout from Firebase, as JWT login is being used (JWT token from Firebase).
    await firebase.FirebaseAuth.instance.signOut();
    user = null;
    notifyListeners();
  }`

### Relevant log output

```shell
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RealmException: User has already been removed
```

**Comment by nielsenko** [comment]
I suggest rearranging the code as follows:
```dart
logout(bool? deleteUser) async {
    final u = user; // <-- use a local final to allow compiler to promote correctly after null check
    if (u == null) {
      return;
    }

    await u.logOut(); // <-- log out before you remove

    if (deleteUser == true) {
      await app.removeUser(u);
    }

    //Logout from Firebase, as JWT login is being used (JWT token from Firebase).
    await firebase.FirebaseAuth.instance.signOut();
    user = null;
    notifyListeners();
  }
```
But I do see what the documentation say, so I will get back on this

**Comment by nirinchev** [comment]
Calling `app.removeUser` will log the user out. Can you clarify what you mean by

> The user is still logged in but removed from the device

How do you determine that the user is logged in?

**Comment by Navil** [comment]
> Calling `app.removeUser` will log the user out. Can you clarify what you mean by
> 
> > The user is still logged in but removed from the device
> 
> How do you determine that the user is logged in?

Maybe it is a messup on my end.

Am am also calling deleteUser. What is the correct order of calling deleteUser and removeUser?

Maybe I got confused there.

**Comment by nirinchev** [comment]
I don't think you need to call both. If you call `deleteUser`, that should remove the user from the local device on top of deleting them on the server.

**Comment by Navil** [comment]
> I don't think you need to call both. If you call `deleteUser`, that should remove the user from the local device on top of deleting them on the server.

Maybe this was the reason for my issue. Eould maybe be good do mention that in the function documentation.
